src/Microsoft.XmlSerializer.Generator/src/Sgen.cs

```diff
@@ -296,6 +296,19 @@ private void GenerateFile(List<string> typeNames, string assemblyName, bool prox
 
                 bool gac = assembly.GlobalAssemblyCache;
                 outputDirectory = outputDirectory == null ? (gac ? Environment.CurrentDirectory : Path.GetDirectoryName(assembly.Location)) : outputDirectory;
+
+                if (!Directory.Exists(outputDirectory))
+                {
+                    if (outputDirectory.EndsWith("\""))
+                    {
+                        outputDirectory = outputDirectory.Remove(outputDirectory.Length - 1);
+                        if (!Directory.Exists(outputDirectory))
+                        {
+                            throw new ArgumentException(SR.Format(SR.ErrDirectoryNotExists, outputDirectory));
+                        }
+                    }
+                }
```

Comment:
BUG: you didn't throw for a non-existing directory that does not end with quotes.

Also, please add comments.

Reply:
What about a leading quotation?